### ISSUE TITLE
Build fails while fetching dependencies

### ISSUE THREAD

**Issue by Pyrohh** [neovim/neovim#2425]
```
~/src/neovim % make deps
...

Removing penlight 1.0.0-1...
Removal successful.
Using https://rocks.moonscript.org/lua_cliargs-2.5-1.src.rock... switching to 'build' mode

Error: Directory busted not found inside archive busted.git
Updating manifest for /home/michael/src/neovim-origin/.deps/usr/lib/luarocks/rocks
lua_cliargs 2.5-1 is now built and installed in /home/michael/src/neovim-origin/.deps/usr (license: MIT <http://opensource.org/licenses/MIT>)

Checking stability of dependencies on the absence of
lua_cliargs 2.3-3...

Removing lua_cliargs 2.3-3...
Removal successful.
ninja: build stopped: subcommand failed.
Makefile:68: recipe for target 'deps' failed
make: *** [deps] Error 1
2|~/src/neovim %
```

I assume it's because of https://github.com/Olivine-Labs/busted/pull/401. The following patch stops cmake complaining about the new dependencies:

```diff
diff --git a/third-party/cmake/BuildLuarocks.cmake b/third-party/cmake/BuildLuarocks.cmake
index d02b193..f1256fe 100644
--- a/third-party/cmake/BuildLuarocks.cmake
+++ b/third-party/cmake/BuildLuarocks.cmake
@@ -30,7 +30,7 @@ endif()
 

 add_custom_command(OUTPUT ${DEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps
   COMMAND ${DEPS_BIN_DIR}/luarocks
-  ARGS build lua_cliargs 2.3-3 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
+  ARGS build lua_cliargs 2.5-1 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
   COMMAND ${DEPS_BIN_DIR}/luarocks
   ARGS build luafilesystem 1.6.3-1 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
   COMMAND ${DEPS_BIN_DIR}/luarocks
@@ -42,7 +42,7 @@ add_custom_command(OUTPUT ${DEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps
   COMMAND ${DEPS_BIN_DIR}/luarocks
   ARGS build lua-term 0.1-1 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
   COMMAND ${DEPS_BIN_DIR}/luarocks
-  ARGS build penlight 1.0.0-1 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
+  ARGS build penlight 1.3.2-2 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
   COMMAND ${DEPS_BIN_DIR}/luarocks
   ARGS build mediator_lua 1.1-3 CC=${DEPS_C_COMPILER} LD=${DEPS_C_COMPILER}
   COMMAND ${DEPS_BIN_DIR}/luarocks
```

However, the build still fails:
```
~/src/neovim % make
ninja  -C .deps
ninja: Entering directory `.deps'
[1/5] Generating usr/bin/busted
FAILED: cd /home/michael/src/neovim/.deps && /home/michael/src/neovim/.deps/usr/bin/luarocks build https://raw.githubusercontent.com/Olivine-Labs/busted/master/busted-scm-0.rockspec CC=/usr/lib/hardening-wrapper/bin/cc LD=/usr/lib/hardening-wrapper/bin/cc /home/michael/src/neovim/.deps/usr/share/lua/5.1/busted/outputHandlers
Warning: falling back to curl - install luasec to get native HTTPS support

Error: Directory busted not found inside archive busted.git
ninja: build stopped: subcommand failed.
Makefile:68: recipe for target 'deps' failed
make: *** [deps] Error 1
2|~/src/neovim %
```

(already did `make distclean` if that matters)

**Comment by marlun** [comment]
I'm getting what I think is the same problem except I'm not using ninja?

```
  inflating: penlight-1.3.2/lua/pl/List.lua
  inflating: penlight-1.3.2/lua/pl/sip.lua
  inflating: penlight-1.3.2/lua/pl/strict.lua
Updating manifest for /Users/marlun/Repos/neovim/.deps/usr/lib/luarocks/rocks
penlight 1.3.2-2 is now built and installed in /Users/marlun/Repos/neovim/.deps/usr (license: MIT/X11)

Checking stability of dependencies on the absence of
penlight 1.0.0-1...

Removing penlight 1.0.0-1...
Removal successful.
Using https://rocks.moonscript.org/lua_cliargs-2.5-1.src.rock... switching to 'build' mode
Updating manifest for /Users/marlun/Repos/neovim/.deps/usr/lib/luarocks/rocks
lua_cliargs 2.5-1 is now built and installed in /Users/marlun/Repos/neovim/.deps/usr (license: MIT <http://opensource.org/licenses/MIT>)

Checking stability of dependencies on the absence of
lua_cliargs 2.3-3...

Removing lua_cliargs 2.3-3...
Removal successful.

Error: Directory busted not found inside archive busted.git
make[3]: *** [usr/bin/busted] Error 1
make[2]: *** [CMakeFiles/busted.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [deps] Error 2
```

**Comment by bfredl** [comment]
I get same error as @marlun

**Comment by jszakmeister** [comment]
I believe the problem is Olivine-Labs/busted@feb5adbe2eb2979bdc65557fe0e3817b19849417.  Luarocks is failing to install correctly from that rockspec for some reason.

**Comment by bfredl** [comment]
Indeed, downloading the rockspec file and reverting that change manually seems to fix the build.

**Comment by jszakmeister** [comment]
I left more information [here](https://github.com/Olivine-Labs/busted/pull/409#issuecomment-92722202).  The short form is that you're supposed to use a url of the form `git+https://`, but that doesn't actually work because there is no git_https module in luarocks.  So it appears we have a series of failures across several projects. :-(

**Comment by bfredl** [comment]
`git+http` seems to work, or I suppose one could download a master tarball from github over pure https if the "s" is important.

**Comment by jszakmeister** [comment]
@bfredl I should mentioned that here, but I did in the busted issue.

**Comment by jszakmeister** [comment]
Fixed for now in d22f2f94047a5ed1e930ee46d0d2e8d6aacfa104.
